internal/webserver/handler-ui.go

```diff
@@ -332,3 +332,65 @@ func (h *handler) serveBookmarkArchive(w http.ResponseWriter, r *http.Request, p
 		log.Printf("error writting response: %s", err)
 	}
 }
+
+// serveEbook is handler for GET /bookmark/:id/ebook
+func (h *handler) serveBookmarkEbook(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	ctx := r.Context()
+
+	// Get bookmark ID from URL
+	strID := ps.ByName("id")
+	id, err := strconv.Atoi(strID)
+	checkError(err)
+	// Get bookmark in database
+	bookmark, exist, err := h.DB.GetBookmark(ctx, id, "")
+	checkError(err)
+
+	if !exist {
+		panic(fmt.Errorf("bookmark not found"))
```

Comment:
I know there are panics all around the API, but we should avoid it unless there's no other way. Just log an error here, return a 404 response and end the function.

Reply:
That's legacy code I had no involvement with, I assume this was just done as the first draft for shiori and just never got updated. It will get fixed in #497

Reply:
The easiest way is use `http.NotFound(r, w)`. #497 is a huge API refactor so don't worry about that.